pkg/storage/segments/store_test.go

```diff
@@ -1,464 +1,254 @@
 // Copyright (C) 2019 Storj Labs, Inc.
 // See LICENSE for copying information.
 
-package segments
+package segments_test
 
 import (
+	"bytes"
 	"context"
+	"crypto/rand"
 	"fmt"
-	"strings"
+	io "io"
+	"io/ioutil"
 	"testing"
-	"time"
+	time "time"
 
-	"github.com/golang/mock/gomock"
-	"github.com/golang/protobuf/ptypes"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vivint/infectious"
 
-	"storj.io/storj/internal/teststorj"
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
 	"storj.io/storj/pkg/eestream"
-	mock_eestream "storj.io/storj/pkg/eestream/mocks"
-	mock_overlay "storj.io/storj/pkg/overlay/mocks"
-	mock_pointerdb "storj.io/storj/pkg/pointerdb/pdbclient/mocks"
-	mock_ecclient "storj.io/storj/pkg/storage/ec/mocks"
-
 	"storj.io/storj/pkg/pb"
-	pdb "storj.io/storj/pkg/pointerdb/pdbclient"
+	ecclient "storj.io/storj/pkg/storage/ec"
 	"storj.io/storj/pkg/storage/meta"
-	"storj.io/storj/pkg/storj"
-)
-
-var (
-	ctx = context.Background()
+	"storj.io/storj/pkg/storage/segments"
+	storj "storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/console"
+	"storj.io/storj/storage"
+	"storj.io/storj/uplink/metainfo"
 )
 
-func TestNewSegmentStore(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	mockOC := mock_overlay.NewMockClient(ctrl)
-	mockEC := mock_ecclient.NewMockClient(ctrl)
-	mockPDB := mock_pointerdb.NewMockClient(ctrl)
-	rs := eestream.RedundancyStrategy{
-		ErasureScheme: mock_eestream.NewMockErasureScheme(ctrl),
-	}
-
-	ss := NewSegmentStore(mockOC, mockEC, mockPDB, rs, 10)
-	assert.NotNil(t, ss)
-}
+// TODO add more error cases
 
 func TestSegmentStoreMeta(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	mockOC := mock_overlay.NewMockClient(ctrl)
-	mockEC := mock_ecclient.NewMockClient(ctrl)
-	mockPDB := mock_pointerdb.NewMockClient(ctrl)
-	rs := eestream.RedundancyStrategy{
-		ErasureScheme: mock_eestream.NewMockErasureScheme(ctrl),
-	}
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
 
-	ss := segmentStore{mockOC, mockEC, mockPDB, rs, 10}
-	assert.NotNil(t, ss)
+	planet, err := testplanet.New(t, 1, 10, 1)
```

Comment:
Do you really need 10 storage nodes for this test? If not reduce the number to the required for the tests. This will reduce the test time.